spotify_player/src/config/mod.rs

```diff
@@ -386,6 +388,34 @@ impl AppConfig {
             ..Default::default()
         }
     }
+
+    /// Returns stdout of `client_id_cmd` if set, otherwise it returns the the value of `client_id`
+    pub fn get_client_id(&self) -> Result<String> {
+        if let Some(cmd) = self.client_id_cmd.clone() {
+            let output = if cfg!(target_os = "windows") {
+                std::process::Command::new("cmd")
+                    .args(["/C", &cmd])
+                    .output()
+            } else {
+                std::process::Command::new("sh")
+                    .arg("-c")
+                    .arg(&cmd)
+                    .output()
+            }?;
```

Comment:
Should we use `config::Command` struct to represent this for consistency (e.g. with previous `player_event_hook_command`)? See https://github.com/aome510/spotify-player/blob/c3e9a4e8b30e00240fb262c89c25de43eb195bdc/spotify_player/src/config/mod.rs#L138-L142

I guess specifying a command string is more convenient, but the above code looks kinda "hacky" to me.

Reply:
Ah I didn't know this existed in the codebase, this does seem like a better idea. 
I could pull out the execution logic for `player_event_hook_command` and implement that on the Command struct